backend/oidc/views.py

```diff
@@ -0,0 +1,71 @@
+import requests
+from django.conf import settings
+from django.contrib import auth
+from django.core.exceptions import SuspiciousOperation
+from django.http import HttpResponse, HttpResponseRedirect, JsonResponse
+from django.views.generic import View
+from mozilla_django_oidc.views import OIDCLogoutView
+from requests.exceptions import HTTPError
+
+from oidc.auth import HelsinkiOIDCAuthenticationBackend
+from oidc.models import OIDCProfile
+from oidc.services import clear_oidc_profile
+
+
+class HelsinkiOIDCLogoutView(OIDCLogoutView):
+    """Override OIDCLogoutView to match the keycloak backchannel logout"""
+
+    def post(self, request):
+        if request.user.is_authenticated:
+            try:
+                oidc_profile = OIDCProfile.objects.get(user=request.user)
+            except OIDCProfile.DoesNotExist:
+                auth.logout(request)
+                return HttpResponseRedirect(self.redirect_url)
+
+            payload = {
+                "id_token_hint": oidc_profile.id_token,
+                "refresh_token": oidc_profile.refresh_token,
+                "client_id": settings.OIDC_RP_CLIENT_ID,
+                "client_secret": settings.OIDC_RP_CLIENT_SECRET,
+            }
+
+            response = requests.post(
+                settings.OIDC_OP_LOGOUT_ENDPOINT,
+                data=payload,
+                verify=self.get_settings("OIDC_VERIFY_SSL", True),
+                timeout=self.get_settings("OIDC_TIMEOUT", None),
+                proxies=self.get_settings("OIDC_PROXY", None),
+            )
+            response.raise_for_status()
```

Comment:
if OIDC_OP_LOGOUT_ENDPOINT is unreachable, is the user unable to log out?

Reply:
Good point! Probably the best solution is just to ignore an erroneous response?

Reply:
Probably the only thing that needs to be done is to log the error in a way that it can be sent to some external monitoring system (is there going to be Sentry or similar in use?).

Reply:
IIRC we don't have sentry in the IBM infra so I guess I will just log this with python logger. Something like:
```
logger.error(f"Helsinki Profile logout was not successful. Reponse status: {response.status}. Response content: {response.content}")
```